wagtail/permission_policies/collections.py

```diff
@@ -94,8 +103,8 @@ def _users_with_perm_filter(self, actions, collection=None):
             collections = collection.get_ancestors(inclusive=True)
             groups = groups.filter(collection_permissions__collection__in=collections)
 
-        # Find all users who are superusers or in any of these groups, and are active
-        return (Q(is_superuser=True) | Q(groups__in=groups)) & Q(is_active=True)
+        # Find all users who are active, and are superusers or in any of these groups
+        return Q(is_active=True) & (Q(is_superuser=True) | Q(groups__in=groups))
```

Comment:
I'm not sure if this actually matters on the database level, but I'm thinking the DB might be able to do some short-circuiting so that the `groups__in` query isn't actually evaluated if `is_active` is `False` or `is_superuser` is `True` now that we put the checks in front.

Reply:
https://www.postgresql.org/docs/current/sql-expressions.html#SYNTAX-EXPRESS-EVAL

> Furthermore, if the result of an expression can be determined by evaluating only some parts of it, then other subexpressions might not be evaluated at all.

so.. it depends

Reply:
Interesting. Looks like the short-circuit can be guaranteed with `CASE`, but we won't be eliminating the number of queries executed by Django so I don't think it's worth the reduced readability